lib/origen/application.rb

```diff
@@ -50,6 +50,8 @@ def inherited(base)
               Origen.register_application(app)
             end
             app.add_lib_to_load_path!
+            # Also blow this cache as a new app has just been added
+            @apps_by_namespace = nil
```

Comment:
This change has no effect on the bug in question, it would still fail with this in place. However, this could potentially be a corner case issue I think if a plugin was loaded manually by an app or something, so took the opportunity to improve it when working in this area.